README.md

```diff
@@ -1,2 +1,91 @@
 # queries
+
 Hosts reusable log queries which are built into a single queries.json file.
+
+## Query database structure
+
+```yaml
+queries:
+  pip-no-matching-distribution:  # also known as query "id", (sort key)
+    pattern: "ERROR: No matching distribution found for"  # required, can be list!
+    # optionals:
+    name: "PIP failed to find package"  # optional (aka 'reason'), id is used instead
```

Comment:
this definitely should be "message"or "reason", not "name", and it should be mandatory, not optional
this is what user will see in the end

Reply:
I prefer `name` due to similarity with ansible, zuul. It is optional because the human-id (key) is used implicitly, otherwise we would have a huge number of cases where we will have to copy/paste the same text between several places: `key`, `id` and `pattern`.

 I will wait for others to comment.

Reply:
I don't see how ansible and zuul come into picture here, if this is a message displayed to user it should be "message".
I don't understand why it should be optional, please elaborate with examples.

Reply:
That was already exemplified by the `src/data/queries.yml` file.